metadata-ingestion/src/datahub/ingestion/source/fivetran/fivetran.py

```diff
@@ -318,43 +325,62 @@ def _get_connection_details_by_id(
         if connection_id in self._connection_details_cache:
             return self._connection_details_cache[connection_id]
 
+        self.report.report_fivetran_rest_api_call_count()
         try:
-            self.report.report_fivetran_rest_api_call_count()
             conn_details = self.api_client.get_connection_details_by_id(connection_id)
-            # Update Cache
-            if conn_details:
-                self._connection_details_cache[connection_id] = conn_details
-
+            self._connection_details_cache[connection_id] = conn_details
             return conn_details
         except Exception as e:
+            logger.debug(
+                f"Failed to get connection details using rest-api for connector_id: {connection_id}. Error: {str(e)}",
+                exc_info=True,
+            )
             self.report.warning(
                 title="Failed to get connection details for Google Sheets Connector",
-                message=f"Exception occurred while getting connection details from Fivetran API. {e}",
-                context=f"connector_id: {connection_id}",
+                message="Exception occurred while getting connection details from Fivetran API",
+                context=f"connector_id: {connection_id}. Error: {str(e)}",
```

Comment:
IIRC there is `exc` arg

```suggestion
                context=f"connector_id: {connection_id}",
                exc=e,
```